test/apps/iodemo/io_demo.cc

```diff
@@ -971,65 +1078,69 @@ class DemoClient : public P2pDemoCommon {
+            if (!op_info[op_id].total_bytes) {
                 continue;
             }
 
-            if (first_print) {
-                first_print = false;
-            } else {
-                // print comma for non-first printouts
-                log << ", ";
+            if (!first_print) {
+                log << ", "; // print comma for non-first operation
             }
+            first_print = false;
 
-            double throughput_mbs = op_info->total_bytes /
+            // Report bandwidth
+            double throughput_mbs = op_info[op_id].total_bytes /
                                     elapsed / (1024.0 * 1024.0);
+            log << io_op_names[op_id] << " " << throughput_mbs << " MB/s";
+            op_info[op_id].total_bytes = 0;
+
+            // Collect min/max among all connections
+            long delta_min = std::numeric_limits<long>::max(), delta_max = 0;
+            for (int server_index = 0; server_index < _server_info.size();
+                 ++server_index) {
+                server_info_t& server_info = _server_info[server_index];
+                long delta_completed = server_info.num_completed[op_id] -
+                                       server_info.prev_completed[op_id];
+                delta_min = std::min(delta_completed, delta_min);
+                delta_max = std::max(delta_completed, delta_max);
+
+                server_info.prev_completed[op_id] =
+                        server_info.num_completed[op_id];
+            }
 
-            log << op_info->num_iters << " "
-                << io_op_names[op_info->op] << "s at "
-                << throughput_mbs << " MB/s";
-
-            // reset for the next round
-            op_info->total_bytes = 0;
-            op_info->num_iters   = 0;
+            // Report delta of min/max/total operations for every connection
+            log << " min:" << delta_min << " max:" << delta_max
+                << " total:" << op_info[op_id].num_iters;
+            op_info[op_id].num_iters = 0;
         }
 
-        if (!first_print) {
-            if (opts().window_size == 1) {
-                log << ", average latency: " << latency_usec << " usec";
-            }
+        log << ", active: " << _active_servers.size();
+
+        if (opts().window_size == 1) {
+            log << ", latency: " << latency_usec << " usec";
         }
     }
 
 private:
-    long                               _num_sent;
-    long                               _num_completed;
-    status_t                           _status;
-    std::map<status_t, std::string>    _status_str;
-    double                             _start_time;
-    unsigned                           _retry;
-    MemoryPool<IoReadResponseCallback> _read_callback_pool;
+    std::vector<server_info_t>              _server_info;
+    std::vector<int>                        _active_servers;
+    std::map<const UcxConnection*, int>     _conn_idx;
+    long                                    _num_sent;
```

Comment:
add print interval

Reply:
@yosefe Do you mean add a comand line option(something like "-p 1.0s") for print interval?